### PR TITLE
refactor(safety-gymnasium): code decoupling

### PR DESCRIPTION
**Pull request by muchvo**
## Description

1. Code decoupling.

## Motivation and Context

Why is this change required? What problem does it solve?
If it fixes an open issue, please link to the issue here.
You can use the syntax `close #15213` if this solves the issue #15213

- [ ] I have raised an issue to propose this change ([required](https://github.com/PKU-MARL/omnisafe/issues) for new features and bug fixes)

## Types of changes

What types of changes does your code introduce? Put an `x` in all the boxes that apply:

- [ ] Bug fix (non-breaking change which fixes an issue)
- [ ] New feature (non-breaking change which adds core functionality)
- [x] Breaking change (fix or feature that would cause existing functionality to change)
- [ ] Documentation (update in the documentation)

## Implemented Tasks

- [ ] Subtask 1
- [ ] Subtask 2
- [ ] Subtask 3

## Checklist

Go over all the following points, and put an `x` in all the boxes that apply.
If you are unsure about any of these, don't hesitate to ask. We are here to help!

- [x] I have read the [CONTRIBUTION](https://omnisafe.readthedocs.io/en/latest/developer/contributing.html) guide. (**required**)
- [x] My change requires a change to the documentation.
- [x] I have updated the tests accordingly. (*required for a bug fix or a new feature*)
- [ ] I have updated the documentation accordingly.
- [ ] I have reformatted the code using `make format`. (**required**)
- [ ] I have checked the code using `make lint`. (**required**)
- [ ] I have ensured `make test` pass. (**required**)

### CODE REVIEW

File: envs/safety-gymnasium/safety_gymnasium/agents/racecar.py [hunk 1]
```diff
@@ -0,0 +1,61 @@
...
+    def is_alive(self):
+        """Racecar runs until timeout."""
+        return True
+
+    def reset(self):
+        """No need to reset anything."""
+
+    def debug(self):
+        """Apply action which inputted from keyboard."""
+
```
Comment (zmsn-2077) [inline envs/safety-gymnasium/safety_gymnasium/agents/racecar.py:50]: unuse line.

File: envs/safety-gymnasium/safety_gymnasium/assets/objects/push_box.py [hunk 2]
```diff
@@ -34,25 +35,38 @@ class PushBox:  # pylint: disable=too-many-instance-attributes
...
             'density': self.density,
             'group': self.group,
             'rgba': self.color,
         }
         return obj
+
+    def _specific_agent_config(self):
+        """Modify the push_box property according to specific agent."""
+        if self.agent.__class__.__name__ == 'Car':
+            # pylint: disable=no-member
```
Comment (zmsn-2077) [inline envs/safety-gymnasium/safety_gymnasium/assets/objects/push_box.py:64]: disable-next

File: envs/safety-gymnasium/safety_gymnasium/bases/base_agent.py [hunk 3]
```diff
@@ -0,0 +1,364 @@
...
+    def apply_action(self, action, noise=None):
+        """Apply an action to the agent."""
+        action = np.array(action, copy=False)  # Cast to ndarray
+
+        # Set action
+        action_range = self.engine.model.actuator_ctrlrange
+
+        self.engine.data.ctrl[:] = np.clip(
+            action, action_range[:, 0], action_range[:, 1]
+        )  # np.clip(action * 2 / action_scale, -1, 1)
```
Comment (zmsn-2077) [inline envs/safety-gymnasium/safety_gymnasium/bases/base_agent.py:205]: Comments that don't make sense.

File: envs/safety-gymnasium/safety_gymnasium/assets/objects/push_box.py [hunk 4]
```diff
@@ -34,25 +35,38 @@ class PushBox:  # pylint: disable=too-many-instance-attributes
...
+            self.size = 0.125  # Box half-radius size
+            self.keepout = 0.125  # Box keepout radius for placement
+            self.density = 0.0005
```
Comment (XuehaiPan) [inline envs/safety-gymnasium/safety_gymnasium/assets/objects/push_box.py:67]:
```suggestion
            # pylint: disable=no-member
            self.size = 0.125  # Box half-radius size
            self.keepout = 0.125  # Box keepout radius for placement
            self.density = 0.0005
            # pylint: enable=no-member
```

**Review by Gaiejj — APPROVED** [review]

**Pull request merged**